packages/utils/lib/errors.unit.test.ts

```diff
@@ -115,5 +115,73 @@ describe('stringifyError', () => {
             const result = stringifyError(new Error('Test'), { pretty: true });
             expect(result.split('\n').length).toBeGreaterThan(1);
         });
+
+        describe('stringifyError performance - NEW VERSION', () => {
+            it('should not significantly regress from baseline performance', () => {
+                // baseline from old implementation (measured in gh actions ci environment)
+                const baseline: Record<string, number> = {
+                    'Simple Error': 0.002882297399999993,
+                    'Axios Error': 0.004207618500000001,
+                    'Boom Error': 0.0029855881999999953
+                };
```

Comment:
i wouldn't worry about specific baseline. Pick one upper bound number that we are we are happy with for 100_000 iterations and let's make sure it doesn't go above it in the future